### PR TITLE
Add a high level description of the activation phase.

### PR DESCRIPTION
**Pull request by dnadales**

### CODE REVIEW

File: formal-spec/update-protocol.tex [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+$2k$ slots before the end of the epoch. This means that endorsements for a given
+proposal for a given epoch are considered up to $4k$ slots before the end of
+that epoch. This is not to say that the endorsements after this cut-off point
+will not be considered, just that they will in the next epoch.
+
+To see why this is required, consider a last endorsement (required to meet the
+threshold) arriving between $4k$ slots and $2k$ slots before the end of an
+epoch, which will happen at slot $s_e$. If the ledger is asked about whether a
+new version will be activated between $s_e - 2k$ and $s_e$, the answer depends
+on the stability of the last endorsement. If it is stable, then replying ``yes''
```
Comment (nkarag) [inline formal-spec/update-protocol.tex]: "If it is stable"  -> Maybe you wanted to say "if it is not stable"

File: formal-spec/update-protocol.tex [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+might not be correct, since we can rollback (switching to another fork), in
+which the last endorsement didn't occur. If it is not stable, then replying
```
Comment (nkarag) [inline formal-spec/update-protocol.tex]: "If it is not stable"  maybe you wanted to say "If it is stable"

File: formal-spec/update-protocol.tex [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+A candidate proposal has an \emph{endorsement period}:
+\begin{itemize}
+\item The end of the endorsement period \textbf{must} coincide with the end of
+  an epoch.
+\item An endorsement period can span multiple epochs.
+\item At the end of an endorsement period, if the candidate proposal has not
+  gathered the endorsements required by the activation threshold, then a new
+  endorsement period might begin.
+\item A candidate proposal has a maximum number of endorsement periods. This
```
[the remark this answers is not in the document]
Comment (dnadales) [reply formal-spec/update-protocol.tex]: > The number of endorsement periods is fixed for all proposals.

I don't get this. If we activate immediately then we don't have any endorsement period, do we?

File: formal-spec/update-protocol.tex [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  number can be zero. Which means that a proposal is discarded as soon as it
+  enters the activation phase. A proposal with a zero activation phases can be
```
[the remark this answers is not in the document]
Comment (dnadales) [reply formal-spec/update-protocol.tex]: You're right! I replaced "discarded" by "adopted".

File: formal-spec/update-protocol.tex [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  used to supersede a proposal that is already in the activation phase. This
+  allows experts pools to overrule a previously approved proposal if a problem
+  is detected after it enters the activation phase.
```
[the remark this answers is not in the document]
Comment (dnadales) [reply formal-spec/update-protocol.tex]: > I understand that the method that we propose to overrule a faulty proposal that is already in the activation phase, is to submit another proposal with a 0 length endorsement period that has a conflict with the former. 

Yes, indeed.

For now I'd say let's see if we can rule the two mechanisms (conflict resolution and overruling) with a single mechanism.

File: formal-spec/update-protocol.tex (hunk 1 above)
Comment (dnadales) [reply formal-spec/update-protocol.tex]: No I think this is correct. However I think what I wrote is wrong: if the endorsement is stable, when we're fine: even if we roll back and switch to a different fork, we won't be able to switch to a fork in which the endorsement didn't occur. The only problem is if the endorsement is not stable.

File: formal-spec/update-protocol.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (nkarag) [reply formal-spec/update-protocol.tex]: Correct. So, it can be  #Endorsement Periods <= 3

File: formal-spec/update-protocol.tex [hunk 6]
```diff
@@ -3,96 +3,110 @@ \section{Update protocol}
...
+need to synchronize prior to the activation of such
+changes, in order to avoid a chain split, due to lack of consensus. Moreover, before launching the upgraded ledger,
+we need to make sure that the security assumptions hold, i.e., that enough honest stake has upgraded. To this end,
+the activation phase consists of a synchronization mechanism, which is called the \emph{endorsement period}
+that ensures that parties will activate changes synchronously and that the upgraded ledger will be secure.
+
+Approved implementations enter the activation phase based on their \emph{priority} and order in a priority queue that we call the \emph{activation queue}. We call the proposals in the activation queue the \emph{approved proposals}. The order of the approved proposals corresponds to the activation priority that we have assigned to them. This priority is metadata-driven and has been confirmed by the community during the Approval phase.
+
+The first approved proposal in the activation queue will enter the endorsement period. There can be \emph{at most one proposal} in the endorsement period at any time.
```
Comment (dnadales) [inline formal-spec/update-protocol.tex:27]: A proposal can't be in a period. I'd say a proposal can be in the activation phase. The activation phase for a proposal consists of multiple endorsement periods.

File: formal-spec/decentralized-updates.tex [hunk 7]
```diff
@@ -73,6 +73,8 @@
 
 \lstset{style=mystyle}
 
+\includeonly{update-protocol}
```
Comment (dnadales) [inline formal-spec/decentralized-updates.tex:76]: Ups, I'll have to remove this.

File: formal-spec/update-protocol.tex [hunk 8]
```diff
@@ -117,22 +137,22 @@ \subsection{Activation phase}
 We call a proposal that is in the endorsement period the \emph{candidate
   proposal}.
 
-If the new candidate proposal is a consensus-update, the nodes can start
-\emph{endorsing} it. An endorsement is a special transaction that is included
-into a block and signals the fact that a node has downloaded and installed the
-software that implements the update.
+If the new candidate proposal is a consensus-update, the block producers can
+start \emph{endorsing} it. An endorsement is a special metadata field in the
+block, and can be set only by the block producers. An endorsement signals the
```
Comment (nkarag) [inline formal-spec/update-protocol.tex:142]: IMO, after Peter's answer, I would leave endorsements as payload in common transactions and deal with the incentives problem properly at a later stage.

File: formal-spec/update-protocol.tex [hunk 9]
```diff
@@ -290,14 +314,14 @@ \subsubsection{Cancellations}
 
 If the candidate proposal already got the required endorsements (which means
 that the cancellation arrived at or later than the slot at which the tally
-occurred, this is $4k$ slots before the end of an epoch), and is waiting to be
-activated, then the cancellation update cannot stop it. In this case, a new
+occurred, this is $2\sigma$ blocks before the end of an epoch), and is waiting to
```
Comment (nkarag) [inline formal-spec/update-protocol.tex:317]: why do you use \sigma and not \kappa (or "k") for blocks?

**Pull request merged**